### ISSUE TITLE
Error on `ADD ${ARGNAME} /` in Dockerfile

### ISSUE THREAD

**Issue by ahwayakchih** [containers/buildah#1938]
I was redirected here from `libpod` issue creator. I'm sorry if i'm creating issue in wrong repository (in which case please point me to the repository where reporting this issue will be ok :).

`ARG` value does not seem to be evaluated when `${ARG}` or `$ARG` is passed as first parameter to the `ADD` command in `Dockerfile`.

**Steps to reproduce the issue:**
1. create `Dockerfile` like this:

```txt
FROM scratch
ARG ALPINE
ADD ${ALPINE} /
CMD ["/bin/sh"]
```

2. download Alpine to local directory:

```sh
curl -o alpine-minirootfs-3.10.3-x86_64.tar.gz http://dl-cdn.alpinelinux.org/alpine/latest-stable/releases/x86_64/alpine-minirootfs-3.10.3-x86_64.tar.gz
```

3. try to build image:

```sh
podman build -t alpine:test --build-arg ALPINE="alpine-minirootfs-3.10.3-x86_64.tar.gz" .
```

**Describe the results you received:**

There's an error because `podman build` does not seem to evaluate arguments:

```txt
STEP 1: FROM scratch
STEP 2: ARG ALPINE
108d1ef2bf4dc3eab22335a0c8247bf77290ad7f063bd3295a26234b9fa7aab7
STEP 3: ADD ${ALPINE} /
Error: error dry-running "ADD ${ALPINE} /": no files found matching "/home/user/test/${ALPINE}": no such file or directory
```

**Describe the results you expected:**

After changing `ADD ${ALPINE} /` line to `ADD alpine-minirootfs-3.10.3-x86_64.tar.gz /` in `Dockerfile` i get correct result:

```txt
STEP 1: FROM scratch
STEP 2: ARG ALPINE
0014e417f8a7ef249ecafab79e055b73447e4f50acd9dca7439a207df1679d13
STEP 3: ADD alpine-minirootfs-3.10.3-x86_64.tar.gz /
53a54476799019d9e4f18d08535e51d4e8edd53ee7e2b4ad522dc36943f833d9
STEP 4: CMD ["/bin/sh"]
STEP 5: COMMIT alpine:test
50a82fe2edc31059b058ca474f3ef43f221cde5a7a5e4c2168282171b94d32b9
```

**Output of `pacman -Q buildah`:**

```
error: package 'buildah' was not found
```

**Output of `pacman -Q podman`:**

```
podman 1.6.1-1
```

**Output of `podman version` if reporting a `podman build` issue:**

```
podman version 1.6.1
```

**Output of `cat /etc/*release`:**

```
LSB_VERSION=1.4
DISTRIB_ID=Anarchy
DISTRIB_RELEASE=rolling
DISTRIB_DESCRIPTION="Anarchy Linux"
NAME="Arch Linux"
PRETTY_NAME="Arch Linux"
ID=arch
BUILD_ID=rolling
ANSI_COLOR="0;36"
HOME_URL="https://www.archlinux.org/"
DOCUMENTATION_URL="https://wiki.archlinux.org/"
SUPPORT_URL="https://bbs.archlinux.org/"
BUG_REPORT_URL="https://bugs.archlinux.org/"
LOGO=archlinux
```

**Output of `uname -a`:**

```
Linux example 5.3.7-arch1-1-ARCH #1 SMP PREEMPT Fri Oct 18 00:17:03 UTC 2019 x86_64 GNU/Linux
```

**Output of `cat /etc/containers/storage.conf`:**

```
# This file is is the configuration file for all tools
# that use the containers/storage library.
# See man 5 containers-storage.conf for more information
# The "container storage" table contains all of the server options.
[storage]

# Default Storage Driver
driver = "overlay"

# Temporary storage location
runroot = "/var/run/containers/storage"

# Primary Read/Write location of container storage
graphroot = "/var/lib/containers/storage"

[storage.options]
# Storage options to be passed to underlying storage drivers

# AdditionalImageStores is used to pass paths to additional Read/Only image stores
# Must be comma separated list.
additionalimagestores = [
]

# Size is used to set a maximum size of the container image.  Only supported by
# certain container storage drivers.
size = ""

# Path to an helper program to use for mounting the file system instead of mounting it
# directly.
#mount_program = "/usr/bin/fuse-overlayfs"

# OverrideKernelCheck tells the driver to ignore kernel checks based on kernel version
override_kernel_check = "true"

# mountopt specifies comma separated list of extra mount options
mountopt = "nodev"

# Remap-UIDs/GIDs is the mapping from UIDs/GIDs as they should appear inside of
# a container, to UIDs/GIDs as they should appear outside of the container, and
# the length of the range of UIDs/GIDs.  Additional mapped sets can be listed
# and will be heeded by libraries, but there are limits to the number of
# mappings which the kernel will allow when you later attempt to run a
# container.
#
# remap-uids = 0:1668442479:65536
# remap-gids = 0:1668442479:65536

# Remap-User/Group is a name which can be used to look up one or more UID/GID
# ranges in the /etc/subuid or /etc/subgid file.  Mappings are set up starting
# with an in-container ID of 0 and the a host-level ID taken from the lowest
# range that matches the specified name, and using the length of that range.
# Additional ranges are then assigned, using the ranges which specify the
# lowest host-level IDs first, to the lowest not-yet-mapped container-level ID,
# until all of the entries have been used for maps.
#
# remap-user = "storage"
# remap-group = "storage"

[storage.options.thinpool]
# Storage Options for thinpool

# autoextend_percent determines the amount by which pool needs to be
# grown. This is specified in terms of % of pool size. So a value of 20 means
# that when threshold is hit, pool will be grown by 20% of existing
# pool size.
# autoextend_percent = "20"

# autoextend_threshold determines the pool extension threshold in terms
# of percentage of pool size. For example, if threshold is 60, that means when
# pool is 60% full, threshold has been hit.
# autoextend_threshold = "80"

# basesize specifies the size to use when creating the base device, which
# limits the size of images and containers.
# basesize = "10G"

# blocksize specifies a custom blocksize to use for the thin pool.
# blocksize="64k"

# directlvm_device specifies a custom block storage device to use for the
# thin pool. Required if you setup devicemapper.
# directlvm_device = ""

# directlvm_device_force wipes device even if device already has a filesystem.
# directlvm_device_force = "True"

# fs specifies the filesystem type to use for the base device.
# fs="xfs"

# log_level sets the log level of devicemapper.
# 0: LogLevelSuppress 0 (Default)
# 2: LogLevelFatal
# 3: LogLevelErr
# 4: LogLevelWarn
# 5: LogLevelNotice
# 6: LogLevelInfo
# 7: LogLevelDebug
# log_level = "7"

# min_free_space specifies the min free space percent in a thin pool require for
# new device creation to succeed. Valid values are from 0% - 99%.
# Value 0% disables
# min_free_space = "10%"

# mkfsarg specifies extra mkfs arguments to be used when creating the base.
# device.
# mkfsarg = ""

# use_deferred_removal marks devicemapper block device for deferred removal.
# If the thinpool is in use when the driver attempts to remove it, the driver 
# tells the kernel to remove it as soon as possible. Note this does not free
# up the disk space, use deferred deletion to fully remove the thinpool.
# use_deferred_removal = "True"

# use_deferred_deletion marks thinpool device for deferred deletion.
# If the device is busy when the driver attempts to delete it, the driver
# will attempt to delete device every 30 seconds until successful.
# If the program using the driver exits, the driver will continue attempting
# to cleanup the next time the driver is used. Deferred deletion permanently
# deletes the device and all data stored in device will be lost.
# use_deferred_deletion = "True"

# xfs_nospace_max_retries specifies the maximum number of retries XFS should
# attempt to complete IO when ENOSPC (no space) error is returned by
# underlying storage device.
# xfs_nospace_max_retries = "0"

# If specified, use OSTree to deduplicate files with the overlay backend
ostree_repo = ""

# Set to skip a PRIVATE bind mount on the storage home directory.  Only supported by
# certain container storage drivers
skip_mount_home = "false"
```

**Comment by TomSweeneyRedHat** [comment]
@ahwayakchih thanks for the problem report and yes this is the correct project to log `podman build` issues against.  This issue was fixed in #1874 which landed in Buildah v1.11.3 which was then vendored into Podman v1.6.2.  You can get the fix by using the code in Podman from GitHub and I think by doing:  `dnf update podman --enablerepo updates-testing`

I'm going to close this Issue for now if you're still having issues after upgrading, please feel free to reopen this.

**Comment by ahwayakchih** [comment]
@TomSweeneyRedHat thanks!
